calico/felix/fetcd.py

```diff
@@ -203,44 +208,25 @@ def watch_etcd(self, update_splitter):
                                                 recursive=True,
                                                 timeout=Timeout(connect=10,
                                                                 read=90),
-                                                check_cluster_id=True)
+                                                check_cluster_uuid=True)
                     _log.debug("etcd response: %r", response)
-                except ConnectTimeoutError:
-                    _log.warning("Connection timeout when trying to  connect "
-                                 "to etcd")
-                    self._reconnect()
-                except ReadTimeoutError:
+                except (ReadTimeoutError, SocketTimeout) as e:
                     # This is expected when we're doing a poll and nothing
-                    # happened.
-                    _log.debug("Read from etcd timed out, retrying.")
-                    self._reconnect()
-                except socket.timeout:
-                    # That this leaks out appears to be an artifact of running
-                    # urllib3 on top of gevent.  Be defensive and reconnect.
-                    _log.warning("Raw socket.timeout leaked out of "
```

Comment:
This is actually the mainline, I think.  My local testing wasn't hitting it because my virtualenv was poisoned by urllib3 1.10.  I've combined it into the ReadTimeout and demoted back to debug level.